### PR TITLE
NR-62325: Added Obfuscation expression and tests

### PR DESCRIPTION
**Pull request by NSSPKrishna**
# Description

This PR includes:
New Feat => Obfuscation expression

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

Please delete options that are not relevant.

- [x] My commit message follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] My code is formatted to [Go standards](https://go.dev/blog/gofmt)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes. Go [here](https://github.com/newrelic/terraform-provider-newrelic#testing) for instructions on running tests locally.

## How to test this change?

```

resource "newrelic_obfuscation_expression" "foo"{
        name = "Testing"
        description = "%[3]s"
        regex = "(^http)"
}

```

### CODE REVIEW

File: newrelic/resource_newrelic_obfuscation_expression.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+				Type:        schema.TypeString,
+				Description: "Name of expression.",
+				Required:    true,
+			},
+			"regex": {
+				Type:        schema.TypeString,
+				Description: "Regex of expression.",
+				Required:    true,
+			},
+			"expression_id": {
```
Comment (mbazhlekova) [inline newrelic/resource_newrelic_obfuscation_expression.go:49]: We can remove this from the schema and just use the ID from Terraform. Customers find it confusing when there are too many IDs.

File: newrelic/resource_newrelic_obfuscation_expression.go [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+func resourceNewRelicObfuscationExpressionRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConfig := meta.(*ProviderConfig)
+	client := providerConfig.NewClient
+	accountID := selectAccountID(providerConfig, d)
+
+	expressionID := d.Id()
+	expression, err := getObfuscationExpressionByID(ctx, client, accountID, expressionID)
+
+	if err != nil {
+		if _, ok := err.(*nrErrors.NotFound); ok {
```
Comment (mbazhlekova) [inline newrelic/resource_newrelic_obfuscation_expression.go:97]: Does this work when you create an obfuscation expression from Terraform then delete in the UI? Terraform should try to recreate the resource.

File: newrelic/resource_newrelic_obfuscation_expression.go [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+	accountID := selectAccountID(meta.(*ProviderConfig), d)
+	expressionID := d.Id()
+
+	_, err := client.Logconfigurations.LogConfigurationsUpdateObfuscationExpressionWithContext(ctx, accountID, updateInput)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(expressionID)
```
Comment (mbazhlekova) [inline newrelic/resource_newrelic_obfuscation_expression.go:145]: No need to setId again on update unless the ID is changing

File: newrelic/resource_newrelic_obfuscation_expression.go (hunk 2 above)
Comment (NSSPKrishna) [reply newrelic/resource_newrelic_obfuscation_expression.go:97]: Thanks for pointing it out, made the required changes for it to work.

File: newrelic/resource_newrelic_obfuscation_expression.go (hunk 3 above)
Comment (NSSPKrishna) [reply newrelic/resource_newrelic_obfuscation_expression.go:145]: Ok. Made the change

**Review by mbazhlekova — APPROVED** [review]

**Pull request merged**